File: backend/app/models/incident.py
from sqlalchemy import Column
from sqlalchemy import Integer
from sqlalchemy import String
from sqlalchemy import Float

from app.database import Base


class Incident(Base):

    __tablename__ = "incidents"

    id = Column(Integer, primary_key=True, index=True)

    severity = Column(String)

    attack_type = Column(String)

    detected_threat = Column(String)

    mitre_id = Column(String)

    confidence = Column(Float)

    risk_score = Column(Integer)

    source_ip = Column(String)

    failed_attempts = Column(Integer)

    summary = Column(String)

File: frontend/src/types/incident.ts
export interface Incident {
  id: number;
  severity: string;
  attack_type: string;
  detected_threat: string;
  mitre_id: string;
  confidence: number;
  risk_score: number;
  source_ip: string | null;
  failed_attempts: number;
  summary: string;
}

export interface Stats {
  total_incidents: number;
  critical: number;
  high: number;
  medium: number;
  low: number;
}